tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -569,6 +569,12 @@ def test_pass_through_of_chunk_shape_generic_data_chunk_iterator(self):
         dset = self.f["test_dataset"]
         self.assertEqual(dset.chunks, chunk_shape)
 
+    def test_pass_through_of_maxshape_on_h5dataset(self):
+        k = 10
+        self.io.write_dataset(self.f, DatasetBuilder('test_dataset', np.arange(k), attributes={}))
+        dset = H5DataIO(self.f['test_dataset'])
+        self.assertEqual(dset.maxshape, (k,))
+
     #############################################
     #  H5DataIO general
     #############################################
```

Comment:
Small nitpick comment - could you move this test to the "H5DataIO general" section below?

```suggestion
    #############################################
    #  H5DataIO general
    #############################################
    
    def test_pass_through_of_maxshape_on_h5dataset(self):
        k = 10
        self.io.write_dataset(self.f, DatasetBuilder('test_dataset', np.arange(k), attributes={}))
        dset = H5DataIO(self.f['test_dataset'])
        self.assertEqual(dset.maxshape, (k,))
```

Reply:
Could you also add a test that fails without the maxshape property. Something like these could go in the [H5DataIOTests section](https://github.com/hdmf-dev/hdmf/blob/49a60df21016fb4da83d64e84e7bc99dac03e94e/tests/unit/test_io_hdf5_h5tools.py#L3664C1-L3667C33):

```python
    def test_dataio_maxshape(self):
        dataio = H5DataIO(data=np.arange(10), maxshape=(None,))
        self.assertEqual(dataio.maxshape, (None,))

    def test_dataio_maxshape_from_data(self):
        dataio = H5DataIO(data=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
        self.assertEqual(dataio.maxshape, (10,))
```